lib/bundles/inspec-compliance/target.rb

```diff
@@ -27,34 +68,19 @@ def self.resolve(target) # rubocop:disable PerceivedComplexity, Metrics/Cyclomat
         profile_fetch_url = target[:url]
         config = {}
       else
-        # check if we have a compliance token
         config = Compliance::Configuration.new
-        if config['token'].nil? && config['refresh_token'].nil?
-          if config['server_type'] == 'automate'
-            server = 'automate'
-            msg = 'inspec compliance login https://your_automate_server --user USER --ent ENT --dctoken DCTOKEN or --token USERTOKEN'
-          elsif config['server_type'] == 'automate2'
-            server = 'automate2'
-            msg = 'inspec compliance login https://your_automate2_server --user USER --token APITOKEN'
-          else
-            server = 'compliance'
-            msg = "inspec compliance login https://your_compliance_server --user admin --insecure --token 'PASTE TOKEN HERE' "
-          end
-          raise Inspec::FetcherFailure, <<~EOF
-
-            Cannot fetch #{uri} because your #{server} token has not been
-            configured.
-
-            Please login using
-
-                #{msg}
-          EOF
-        end
-
+        check_compliance_token(config)
         # verifies that the target e.g base/ssh exists
         profile = Compliance::API.sanitize_profile_name(uri)
-        if !Compliance::API.exist?(config, profile)
+        # Call profiles directly instead of exist? to capture the results
+        # so we can access the upstream sha256 from the results.
+        _msg, profile_result = Compliance::API.profiles(config, profile)
+        if profile_result.empty?
           raise Inspec::FetcherFailure, "The compliance profile #{profile} was not found on the configured compliance server"
+        else
+          # If version wasn't specified the results are version sorted, so grab the first result.
+          # If version was specified, it will be the first and only result.
+          profile_info = profile_result[0]
```

Comment:
What happens if we have more then one result. Seems unreliable to depend on the first entry since an order from the server is not guaranteed. We do not want to have different profiles running for the same identifier.

Reply:
This is a fair question, especially for legacy Compliance or Automate 1. For Automate 2, the sql results for namespaceProfilesAll and namespaceProfilesByName order the query results by profile name and then descending version. However, since we are using an unversioned api endpoint, there's no guarantee the behavior would not change. I could sort the profiles before returning them here: https://github.com/itmustbejj/inspec/blob/60d3a3746b3a03da4354f715f85ebf5cb3ddc658/lib/bundles/inspec-compliance/api.rb#L81

A2 code sorting profiles for reference:
https://github.com/chef/a2/blob/master/components/compliance-service/profiles/db/store.go#L498-L502
https://github.com/chef/a2/blob/master/components/compliance-service/profiles/db/store.go#L464
https://github.com/chef/a2/blob/master/components/compliance-service/profiles/db/store.go#L481